### PR TITLE
Add previous service case reference to Envelope

### PR DESCRIPTION
**Pull request by lgonczar**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-588

### Change description ###

Add previous service case reference to Envelope.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/model/Envelope.java [hunk 1]
```diff
@@ -24,6 +25,7 @@
     public Envelope(
         @JsonProperty(value = "id", required = true) String id,
         @JsonProperty(value = "case_ref", required = true) String caseRef,
+        @JsonProperty("previous_service_case_ref") String previousServiceCaseRef,
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/model/Envelope.java]: `required=false` for clarity?

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/model/Envelope.java [hunk 2]
```diff
@@ -11,6 +11,7 @@
 
     public final String id;
     public final String caseRef;
+    public final String previousServiceCaseRef;
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/model/Envelope.java]: might be good to add a note here clarifying that it's not the 'previous ServiceCaseRef', it's the 'previous service caseRef'.

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/model/Envelope.java]: or rename the field altogether

Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/model/Envelope.java]: This "previous service case reference" is a convention that is shared with Exela and CCD, so I'd rather not rename it. I'll add a comment.

Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/model/Envelope.java]: Maybe I should call it legacyId, as that's how we normally refer to it?

Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/model/Envelope.java]: `legacyCaseRef` it is

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/model/Envelope.java (hunk 1 above)
Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/model/Envelope.java]: Idea complains about redundant parameter if I do

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/model/Envelope.java]: people > intellij ;)

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/model/Envelope.java [hunk 3]
```diff
@@ -24,6 +25,7 @@
     public Envelope(
         @JsonProperty(value = "id", required = true) String id,
         @JsonProperty(value = "case_ref", required = true) String caseRef,
+        @JsonProperty("previous_service_case_ref") String legacyCaseRef,
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/model/Envelope.java]: are there any changes made in the processor yet?
If not, maybe we could have `legacy_case_ref` in json?

Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/model/Envelope.java]: The processor is already passing this field. I'll change the name as the last thing in this story.

**Pull request merged**